packages/proxy/lib/http/util/cookies.ts

```diff
@@ -3,13 +3,49 @@ import type Debug from 'debug'
 import { URL } from 'url'
 import { cors } from '@packages/network'
 import { AutomationCookie, Cookie, CookieJar, toughCookieToAutomationCookie } from '@packages/server/lib/util/cookies'
+import { calculateSiteContext } from './top-simulation'
+import type { RequestCredentialLevel, RequestResourceType } from '../../types'
 
 interface RequestDetails {
   url: string
   isAUTFrame: boolean
   needsCrossOriginHandling: boolean
 }
 
+export const shouldAttachAndSetCookies = (requestUrl: string, AUTUrl: string | undefined, resourceType?: RequestResourceType, credentialLevel?: RequestCredentialLevel): boolean => {
+  if (!AUTUrl) return false
+
+  const siteContext = calculateSiteContext(requestUrl, AUTUrl)
+
+  switch (resourceType) {
+    case 'fetch':
+      // never attach cookies regardless of siteContext if omit is optioned
+      if (credentialLevel === 'omit') {
+        return false
+      }
+
+      // if the siteContext is same-origin and the credential status is undefined, same-origin, or include, attach cookies.
```

Comment:
Is this comment still correct?  This doesn't seem to match what is happening in the if statement below

'If the siteContext is same-origin **or** the credential level equals include.'

Reply:
I think the comment just stinks 😄 . I updated it [here](https://github.com/cypress-io/cypress/pull/23880/files#diff-8bcc53731d8dc8219a282445e0ae5ae47e2884d0fb28ead3eef2b488819d1bf0R42)